templates/tapestry.js

```diff
@@ -1349,9 +1351,20 @@ function tapestryTool(config){
             .attr("width", d => Math.min(getRadius(d) * 2 + 48, 400))
             .attr("height", d => getRadius(d) * 2)
             .attr("x", d => -(Math.min(getRadius(d) * 2 + 48, 400) / 2))
-            .attr("y", d => -(getRadius(d) * 3 + 27.5 + 20))
+            .attr("y", d => {
+                const tooltipHeight = getRadius(d) * 2.5
+                const onBottom = (d.y - tapestry.getTapestryDimensions().startY < tooltipHeight 
+                    || $(window).scrollTop() > tooltipHeight) && d.x > 0 && d.y > 0
+                return onBottom ? getRadius(d) + 27.5 + 5 : -(getRadius(d) * 3 + 27.5 + 20)
+            })
             .append("xhtml:div")
             .attr("class", "tapestry-tooltip")
+            .style("align-items", d => {
+                const tooltipHeight = getRadius(d) * 2.5
+                const onBottom = (d.y - tapestry.getTapestryDimensions().startY < tooltipHeight 
+                    || $(window).scrollTop() > tooltipHeight) && d.x > 0 && d.y > 0
```

Comment:
I think here, we should use the scrollTop of the tooltip element itself, not the scrollTop of window (which isn't really relevant)

Reply:
Since we're using the y position of the node in the window, the $(window).scrollTop() is the value we need to compare against the position to determine where the tooltip should render. My testing for the tooltip's scrollTop value was always 0, because you don't scroll within that element